File: src/Aircraft.Domain/Maintenance.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace Aircraft.Domain
{
    public class Maintenance
    {
        public Guid Id { get; set; }
        public UserProfile User { get; set; }
        public Guid UserId { get; set; } 
        public string Description { get; set; }
        public bool Status { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public IEnumerable<Stage> Stages { get; set; } 
       
    }
}

File: src/Aircraft.Web/web/src/app/models/maintenance.ts
import { Stage } from "./stage";

export class Maintenance {
    userId : string;
    description : string;
    createdAt : Date;
    status : boolean;
    stage: Stage[];

    constructor(userId : string, description : string, createdAt : Date,  status : boolean,stage: Stage[]) {
        this.userId = userId;
        this.description = description;
        this.createdAt = createdAt;
        this.status = status;
        this.stage = stage;
    }

}